airflow/www/api/experimental/endpoints.py

```diff
@@ -44,16 +49,34 @@ def trigger_dag(dag_id):
     if 'conf' in data:
         conf = data['conf']
 
+    execution_date = None
+    if 'exec_date' in data:
+        execution_date = data['exec_date']
+
+        # Convert string datetime into actual datetime
+        try:
+            execution_date = datetime.strptime(execution_date,
+                                               '%Y-%m-%dT%H:%M:%S')
+        except ValueError:
+            error_message = (
+                'Given execution date, {}, could not be identified '
+                'as a date. Example date format: 2015-11-16T14:34:15'
+                .format(execution_date))
+            _log.info(error_message)
+            response = jsonify({'error': error_message})
+            response.status_code = 400
+
+            return response
+
     try:
-        dr = trigger.trigger_dag(dag_id, run_id, conf)
+        dr = trigger.trigger_dag(dag_id, run_id, conf, execution_date)
     except AirflowException as err:
-        logging.error(err)
```

Comment:
Ah, that's my mistake. In the other endpoints I'm adding, I put the logging in the method in api/common so if it's called from elsewhere and isn't logged there, it still gets logged. I then removed the logging from here so it wasn't logged twice and did the same thing to this one even though I hadn't added the logging in trigger_dag.py. Given that I'll be updating this either way, would you prefer it here or in trigger_dag.py?